src/input/keyboard/mod.rs

```diff
@@ -249,12 +251,166 @@ impl<'a> KeysymHandle<'a> {
             .key_get_syms_by_level(self.keycode, self.state.key_get_layout(self.keycode), 0)
     }
 
+    /// Get the raw latin keysym or fallback to current raw keysym.
+    ///
+    /// This method is handy to implement layout agnostic bindings. Keep in mind that
+    /// it could be not-ideal to use just this function, since some layouts utilize non-standard
+    /// shift levels and you should look into [`Self::modified_sym`] first.
+    ///
+    /// The `None` is returned when the underlying keycode doesn't produce a valid keysym.
+    pub fn raw_latin_sym_or_raw_current_sym(&self) -> Option<Keysym> {
+        let effective_layout = Layout(self.state.key_get_layout(self.keycode));
+        // NOTE: There's always a keysym in the current layout given that we have modified_sym.
+        let base_sym = *self.raw_syms().first()?;
```

Comment:
I see xkb has `key_get_one_sym` for when you only want a single keysym rather than multiple (it returns `NoSymbol` in case there would be multiple), but I don't see a corresponding `by_level` function. Maybe it would make sense to replicate this here by only proceeding when `raw_syms()` has exactly one element. But also it's probably fine as is?

Reply:
it's fine as is, you need api for `layout + level`, the thing I use is basically the only one available.